server/channelserver/handlers_quest.go

```diff
@@ -195,23 +196,50 @@ func handleMsgMhfEnumerateQuest(s *Session, p mhfpacket.MHFPacket) {
 	bf := byteframe.NewByteFrame()
 	bf.WriteUint16(0)
 
-	rows, _ := s.server.db.Query("SELECT id, COALESCE(max_players, 4) AS max_players, quest_type, quest_id, COALESCE(mark, 0) AS mark FROM event_quests ORDER BY quest_id")
+	currentTime := time.Now()
+	var tableName = "event_quests"
+	/* This is an example of how I have to test and avoid issues, might be a thing later?
+ 	if _config.ErupeConfig.RealClientMode <= _config.F5 {
+		tableName = "event_quests_older"
+	}
+ 	*/
+	// Check the event_quests table to load the quests with rotation system
+	rows, err := s.server.db.Query("SELECT id, COALESCE(max_players, 4) AS max_players, quest_type, quest_id, COALESCE(mark, 0) AS mark, start_time, active_duration, inactive_duration FROM " + tableName + "")
+	if err != nil {
+		return
+	}
+	defer rows.Close()
+
 	for rows.Next() {
-		data, err := makeEventQuest(s, rows)
+		var id, mark uint32
+		var questId, activeDuration, inactiveDuration int
+		var maxPlayers, questType uint8
+		var startTime time.Time
+		err := rows.Scan(&id, &maxPlayers, &questType, &questId, &mark, &startTime, &activeDuration, &inactiveDuration)
 		if err != nil {
 			continue
-		} else {
-			if len(data) > 896 || len(data) < 352 {
+		}
+
+		// Calculate the rotation time based on start time, active duration, and inactive duration
+		rotationTime := startTime.Add(time.Duration(activeDuration+inactiveDuration) * 24 * time.Hour)
+		if currentTime.After(rotationTime) {
+			// The rotation time has passed, update the start time and reset the rotation
+			_, err := s.server.db.Exec("UPDATE "+tableName+" SET start_time = $1 WHERE quest_id = $2", rotationTime, questId)
```

Comment:
Same here with this line. Although I think just hardcoding it would be better unless the G7 event quest format is that much different.